connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -271,17 +256,34 @@ private static Connector instantiateConnector(Class<? extends Connector> connCla
         return result;
     }
 
-    public void stopConnector(String connName) {
+    public void stopConnectors() {
+        stopConnectors(new HashSet<>(connectors.keySet()));
+    }
+
+    public Collection<String> stopConnectors(Collection<String> connectors) {
+        final List<String> stopped = new ArrayList<>(connectors.size());
+        for (String connector: connectors) {
+            if (stopConnector(connector)) {
+                stopped.add(connector);
+            }
+        }
+        return stopped;
+    }
+
+    public boolean stopConnector(String connName) {
```

Comment:
Not part of this patch, but this name seems a little misleading. In addition to stopping the connector, we also remove it from the connectors that are managed by the Worker. I'm wondering if it makes sense to separate the notion of connector _installation_? Maybe that would get around some of the awkwardness with removing the connector when we're just restarting it and needing the set copy in `stopConnectors()` above.